crates/store/re_chunk/src/chunk.rs

```diff
@@ -240,6 +240,78 @@ impl Chunk {
             .sum()
     }
 
+    /// The cumulative number of events in this chunk for each _unique_ timestamp.
+    ///
+    /// I.e. how many _component batches_ ("cells") were logged in total at each timestamp?
+    ///
+    /// Keep in mind that a timestamp can appear multiple times in a [`Chunk`].
+    /// This method will do a sum accumulation to account for these cases (i.e. every timestamp in
+    /// the returned vector is guaranteed to be unique).
+    #[inline]
+    pub fn num_events_cumulative_per_unique_time(
+        &self,
+        timeline: &Timeline,
+    ) -> Vec<(TimeInt, u64)> {
+        re_tracing::profile_function!();
+
+        let iter_times = || {
+            if self.is_static() {
+                arrow2::Either::Left(std::iter::repeat(TimeInt::STATIC))
+            } else {
+                let times = self.timelines.get(timeline).map_or_else(
+                    || arrow2::Either::Left(std::iter::empty()),
```

Comment:
I suggest we early-out when we don't find the timeline. That way the iterator becomes simpler, and therefor faster